
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://sylph-docs.github.io/old/Integrating-taxonomic-information-with-sylph/">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>Integrating taxonomic information with sylph - Documentation for sylph - ultrafast, precise metagenomic profiling</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#sylph-utils-scripts-for-integrating-taxonomic-information" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-header__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Documentation for sylph - ultrafast, precise metagenomic profiling
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Integrating taxonomic information with sylph
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-nav__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Documentation for sylph - ultrafast, precise metagenomic profiling
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-can-cannot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    What sylph can and cannot do
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../install%2Bquickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../pre%E2%80%90built-databases/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pre-built databases
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-cookbook/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cookbook
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Output format
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../5%E2%80%90minute-sylph-tutorial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 1 - 5 minute introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../taxonomic-profiling-tutorial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 2 - taxonomic profiling with GTDB
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Taxonomy integration (sylph-tax)
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            Taxonomy integration (sylph-tax)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-tax/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sylph-tax
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-tax-quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-tax-commands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Subcommand information
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-tax-output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    .sylphmpa output format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sylph-tax-custom-taxonomies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Custom taxonomies
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#sylph-utils-scripts-for-integrating-taxonomic-information" class="md-nav__link">
    <span class="md-ellipsis">
      sylph-utils - scripts for integrating taxonomic information
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-to-generate-profiles-with-taxonomic-information-using-sylph-utils" class="md-nav__link">
    <span class="md-ellipsis">
      How to generate profiles with taxonomic information using sylph-utils
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metaphlan-like-or-cami-like-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      MetaPhlAn-like or CAMI-like outputs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#custom-taxonomies-and-how-it-works" class="md-nav__link">
    <span class="md-ellipsis">
      Custom taxonomies and how it works
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-taxonomy-metadata-from-refseq" class="md-nav__link">
    <span class="md-ellipsis">
      Creating taxonomy metadata from RefSeq?
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Integrating taxonomic information with sylph</h1>

<blockquote>
<p>[!WARNING]
sylph-utils has been replaced with <a href="https://github.com/bluenote-1577/sylph-tax">sylph-tax</a>. sylph-utils still works but will not be updated anymore. </p>
</blockquote>
<h2 id="sylph-utils-scripts-for-integrating-taxonomic-information"><a href="https://github.com/bluenote-1577/sylph-utils">sylph-utils</a> - scripts for integrating taxonomic information</h2>
<p>The <a href="https://github.com/bluenote-1577/sylph-utils">sylph-utils</a> repository contains scripts/metadata for integrating taxonomy information into sylph's output. Details are given below.</p>
<h2 id="how-to-generate-profiles-with-taxonomic-information-using-sylph-utils">How to generate profiles with taxonomic information using sylph-utils</h2>
<p>By default, sylph's TSV outputs contain no taxonomic information. However, sylph supports integrating a few major genome databases for taxonomic profiles. The pre-indexed versions of a few databases are available <a href="https://github.com/bluenote-1577/sylph/wiki/Pre%E2%80%90built-databases">here</a>.</p>
<p>In the <a href="https://github.com/bluenote-1577/sylph-utils">sylph-utils</a> repository, the <code>sylph_to_taxprof.py</code> script can be used as follows</p>
<div class="highlight"><pre><span></span><code>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/bluenote-1577/sylph-utils
sylph<span class="w"> </span>profile<span class="w">  </span>gtdb-r220-c200-dbv1.syldb<span class="w"> </span>sample.sylsp<span class="w"> </span>-o<span class="w"> </span>result.tsv
python<span class="w"> </span>sylph-utils/sylph_to_taxprof.py<span class="w"> </span>-s<span class="w"> </span>result.tsv<span class="w"> </span>-m<span class="w"> </span>sylph-utils/prokaryote/gtdb_r220_metadata.tsv.gz
ls<span class="w"> </span>*.sylphmpa
</code></pre></div>
<p>The <code>gtdb_r220_metadata.tsv.gz</code> file links the GTDB taxonomy to each genome. It groups all the results by sample name and outputs a file called <code>sample.sylphmpa</code> with the format shown below.  </p>
<h2 id="metaphlan-like-or-cami-like-outputs">MetaPhlAn-like or CAMI-like outputs</h2>
<p><code>*.sylphmpa</code> files look like this: </p>
<div class="highlight"><pre><span></span><code>#SampleID       biofilm_reads/SRR24442552_1.fastq.gz
clade_name     relative_abundance      sequence_abundance      ANI (if strain-level)
d__Bacteria     99.99999999999997       3.4775000000000014      NA
d__Bacteria|p__Actinomycetota   1.7266  0.0414  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia 1.3727  0.0313  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales     1.3727  0.0313  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae       1.3727  0.0313  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae|g__Casp-actino5       1.0267  0.0172  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae|g__Casp-actino5|s__Casp-actino5       1.0267  0.0172  NA
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae|g__Casp-actino5|s__Casp-actino5|t__GCA_017859785.1    1.0267  0.0172  98.54
d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae|g__Ilumatobacter      0.346   0.0141  NA
....
</code></pre></div>
<blockquote>
<p>[!TIP]
This is a valid TSV file, but rows prefixed with <code>#</code> are comments.
You can read <code>.sylphmpa</code> files with pandas in python like <code>pd.read_csv('output.sylphmpa',sep='\t', comment='#')</code>. </p>
</blockquote>
<p>There are five important columns:</p>
<ol>
<li><code>clade_name</code>: A string like <code>d__Bacteria|p__Actinomycetota|c__Acidimicrobiia|o__Acidimicrobiales|f__Ilumatobacteraceae</code> that describes the clade. <code>t__STRAIN</code> represents the exact genome identifier. </li>
<li><code>relative_abundance</code>: the taxonomic relative abundance of the clade</li>
<li><code>sequence_abundance</code>: the sequence abundance of the clade, i.e. the % of reads assigned</li>
<li><code>ANI</code>: this is <code>NA</code> except for at the strain level (<code>t__strain</code>). Otherwise it is sylph's ANI estimate. </li>
<li><code>Coverage</code>: (new in v0.2 of <code>sylph_to_taxprof.py</code>) This is the <code>Eff_cov</code> or <code>True_cov</code> column of sylph's output.</li>
</ol>
<blockquote>
<p>[!TIP]
In v0.2 of <code>sylph_to_taxprof.py</code>, viral-host information is available for IMG/VR 4.1. The <code>-a</code> option adds a new column in the .sylphmpa files associating viral genomes to their hosts. For example, a row can look like : </p>
<p><code>r__Duplodnaviria|k__Heunggongvirae|p__Uroviricota|c__Caudoviricetes|||||t__IMGVR_UViG_2503982007_000001 ...    d__Bacteria;p__Firmicutes;c__Bacilli;o__Staphylococcales;f__Staphylococcaceae;g__Staphylococcus;s__Staphylococcus epidermidis</code> </p>
<p>where IMGVR_UVIG_2503982007's host is Staphylococcus epidermidis.</p>
</blockquote>
<h2 id="custom-taxonomies-and-how-it-works">Custom taxonomies and how it works</h2>
<p>The <code>sylph_to_taxprof.py</code> file is quite simple. If you're serious about using a new taxonomy, it should be short enough to read. </p>
<p>Briefly: the <code>gtdb_r220_metadata.tsv</code> file (can be gzipped or not) specified by the <code>-m</code> option is a file with two columns. The first column is the name of your fasta file, and the second column is a taxonomy string that looks like <code>d__Archaea;p__Methanobacteriota_B;c__Thermococci;o__Thermococcales;f__Thermococcaceae;g__Thermococcus_A;s__Thermococcus_A alcaliphilus</code>.</p>
<p>Let's say you want to add <code>genome1.fa</code> and <code>genome2.fa</code>, two new MAGs, to the GTDB taxonomy. You can change the <code>gtdb_r220_metadata.tsv</code> file, to look like this by appending two new lines:</p>
<p><div class="highlight"><pre><span></span><code>...
...
GCA_945889495.1 d__Bacteria;p__Desulfobacterota_B;c__Binatia;o__UBA9968;f__UBA9968;g__DP-20;s__DP-20 sp945889495
GCA_934500585.1 d__Bacteria;p__Bacillota_A;c__Clostridia;o__UBA1381;f__UBA1381;g__RQCD01;s__RQCD01 sp008668455
genome1.fa d__Archaea;(...);s__My new species name`
genome2.fa d__Bacteria;(...);g__My_genus_name;s__My species name2`
</code></pre></div>
and use it for the <code>-m</code> option in the python script. Don't add the <code>t__STRAIN</code> line; the script does it automatically. </p>
<blockquote>
<p>[!WARNING] 
For Genbank/RefSeq assemblies, filenames have to be dealt with carefully.</p>
<p><strong>If <code>_genomic</code> or <code>_ASM</code> is in your genome file name, e.g. GCF_002863645.1_ASM286364v1_genomic.fna.gz, use "GCF_002863645.1", instead of the whole file name in the first column.</strong></p>
</blockquote>
<h2 id="creating-taxonomy-metadata-from-refseq">Creating taxonomy metadata from RefSeq?</h2>
<p>See <a href="https://github.com/bluenote-1577/sylph/issues/14#issuecomment-2313259637">this discussion thread</a>.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.56ea9cef.min.js"></script>
      
    
  </body>
</html>