
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://sylph-docs.github.io/5%E2%80%90minute-sylph-tutorial/">
      
      
        <link rel="prev" href="../Output-format/">
      
      
        <link rel="next" href="../taxonomic-profiling-tutorial/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>Tutorial 1 - 5 minute introduction - Documentation for sylph - ultrafast, precise metagenomic profiling</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#5-minute-tutorial" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-header__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Documentation for sylph - ultrafast, precise metagenomic profiling
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Tutorial 1 - 5 minute introduction
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-nav__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Documentation for sylph - ultrafast, precise metagenomic profiling
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-can-cannot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    What sylph can and cannot do
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../install%2Bquickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../pre%E2%80%90built-databases/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pre-built databases
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-cookbook/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cookbook
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Output format
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Tutorial 1 - 5 minute introduction
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Tutorial 1 - 5 minute introduction
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#5-minute-tutorial" class="md-nav__link">
    <span class="md-ellipsis">
      5 minute tutorial
    </span>
  </a>
  
    <nav class="md-nav" aria-label="5 minute tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sketching" class="md-nav__link">
    <span class="md-ellipsis">
      Sketching
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#querying" class="md-nav__link">
    <span class="md-ellipsis">
      Querying
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#profiling-vs-ani-querying-whats-the-difference" class="md-nav__link">
    <span class="md-ellipsis">
      Profiling vs ANI querying. What's the difference?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#real-profiling-against-a-database" class="md-nav__link">
    <span class="md-ellipsis">
      Real profiling against a database
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../taxonomic-profiling-tutorial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 2 - taxonomic profiling with GTDB
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Taxonomy integration (sylph-tax)
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            Taxonomy integration (sylph-tax)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sylph-tax
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-commands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Subcommand information
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    .sylphmpa output format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-custom-taxonomies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Custom taxonomies
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#5-minute-tutorial" class="md-nav__link">
    <span class="md-ellipsis">
      5 minute tutorial
    </span>
  </a>
  
    <nav class="md-nav" aria-label="5 minute tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sketching" class="md-nav__link">
    <span class="md-ellipsis">
      Sketching
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#querying" class="md-nav__link">
    <span class="md-ellipsis">
      Querying
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#profiling-vs-ani-querying-whats-the-difference" class="md-nav__link">
    <span class="md-ellipsis">
      Profiling vs ANI querying. What's the difference?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#real-profiling-against-a-database" class="md-nav__link">
    <span class="md-ellipsis">
      Real profiling against a database
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Tutorial 1 - 5 minute introduction</h1>

<h2 id="5-minute-tutorial">5 minute tutorial</h2>
<h3 id="sketching">Sketching</h3>
<p>After installation, clone this repository if you have not done so and run the following.</p>
<div class="highlight"><pre><span></span><code>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/bluenote-1577/sylph
<span class="nb">cd</span><span class="w"> </span>sylph

<span class="c1">## install sylph. see installation instructions</span>
<span class="c1"># cargo install --path . --root ~/.cargo</span>
<span class="c1">## OR</span>
<span class="c1"># conda install -c bioconda sylph</span>

<span class="c1"># sketch reads and genomes. fastq -&gt; samples, fasta -&gt; queries</span>
sylph<span class="w"> </span>sketch<span class="w"> </span>test_files/o157_reads.fastq<span class="w"> </span>test_files/e.coli*.fa<span class="w"> </span>-o<span class="w"> </span>database
</code></pre></div>
<p>There are two types of files: <code>*.syldb</code> and <code>*.sylsp</code>.</p>
<ul>
<li><strong>FASTQ files</strong> are treated as <strong>samples</strong> and turn into <code>*.sylsp</code> </li>
<li><strong>FASTA files</strong> are treated as <strong>genomes</strong> and turn into <code>*.syldb</code></li>
<li>For other options, such as paired-end reads, see the <a href="../sylph-cookbook/">cookbook</a></li>
</ul>
<p>Genomes are aggregated into one <code>syldb</code> file, and each genome is queried against all <code>sylsp</code>s. </p>
<h3 id="querying">Querying</h3>
<div class="highlight"><pre><span></span><code><span class="c1"># query for ANI </span>
sylph<span class="w"> </span>query<span class="w"> </span>o157_reads.fastq.sylsp<span class="w"> </span>database.syldb

<span class="c1"># ALTERNATIVE: lazy containment without pre-sketching also works </span>
<span class="c1"># sylph query test_files/*</span>
</code></pre></div>
<p>The o157_reads.fastq is a "metagenomic sample" containing only E. coli O157 with 1x coverage and 95% identity reads (i.e. 5% error). We query the reads and the database consisting of multiple E. coli genomes. </p>
<p>You'll see something like the following</p>
<div class="highlight"><pre><span></span><code>Sample_file<span class="w"> </span>Genome_file<span class="w"> </span>Adjusted_ANI<span class="w">    </span>Eff_cov<span class="w"> </span>ANI_5-95_percentile<span class="w"> </span>Eff_lambda<span class="w">  </span>Lambda_5-95_percentile<span class="w">  </span>Median_cov<span class="w">  </span>Mean_cov_geq1<span class="w">   </span>Containment_ind<span class="w"> </span>Naive_ANI<span class="w">   </span>Contig_name
../test_files/o157_reads.fastq<span class="w">  </span>../test_files/e.coli-o157.fasta<span class="w"> </span><span class="m">99</span>.73<span class="w">   </span><span class="m">0</span>.360<span class="w">   </span><span class="m">99</span>.61-99.91<span class="w"> </span><span class="m">0</span>.360<span class="w">   </span><span class="m">0</span>.33-0.38<span class="w">   </span><span class="m">1</span><span class="w">   </span><span class="m">1</span>.187<span class="w">   </span><span class="m">6208</span>/21899<span class="w">  </span><span class="m">96</span>.02<span class="w">   </span>NZ_CP017438.1<span class="w"> </span>Escherichia<span class="w"> </span>coli<span class="w"> </span>O157:H7<span class="w"> </span>strain<span class="w"> </span><span class="m">2159</span><span class="w"> </span>chromosome,<span class="w"> </span><span class="nb">complete</span><span class="w"> </span>genome
../test_files/o157_reads.fastq<span class="w">  </span>../test_files/e.coli-EC590.fasta<span class="w">    </span><span class="m">98</span>.25<span class="w">   </span><span class="m">0</span>.319<span class="w">   </span><span class="m">98</span>.06-98.55<span class="w"> </span><span class="m">0</span>.319<span class="w">   </span><span class="m">0</span>.29-0.34<span class="w">   </span><span class="m">1</span><span class="w">   </span><span class="m">1</span>.172<span class="w">   </span><span class="m">3122</span>/19330<span class="w">  </span><span class="m">94</span>.29<span class="w">   </span>NZ_CP016182.2<span class="w"> </span>Escherichia<span class="w"> </span>coli<span class="w"> </span>strain<span class="w"> </span>EC590<span class="w"> </span>chromosome,<span class="w"> </span><span class="nb">complete</span><span class="w"> </span>genome
../test_files/o157_reads.fastq<span class="w">  </span>../test_files/e.coli-K12.fasta<span class="w">  </span><span class="m">98</span>.16<span class="w">   </span><span class="m">0</span>.327<span class="w">   </span><span class="m">97</span>.96-98.47<span class="w"> </span><span class="m">0</span>.327<span class="w">   </span><span class="m">0</span>.29-0.35<span class="w">   </span><span class="m">1</span><span class="w">   </span><span class="m">1</span>.171<span class="w">   </span><span class="m">3114</span>/19485<span class="w">  </span><span class="m">94</span>.26<span class="w">   </span>NC_007779.1<span class="w"> </span>Escherichia<span class="w"> </span>coli<span class="w"> </span>str.<span class="w"> </span>K-12<span class="w"> </span>substr.<span class="w"> </span>W3110,<span class="w"> </span><span class="nb">complete</span><span class="w"> </span>sequence
</code></pre></div>
<p>These are statistics for each genome against our reads. The ANI can be interpreted as nearest-neighbour ANI searching, i.e., "what is the highest ANI between the genomes in my sample and the reference genome?". See <a href="../Output-format/">here</a> for a detailed output explanation. </p>
<ol>
<li>The 3rd column gives the coverage adjusted ANI between the genome and this sample. </li>
<li>The second last column is the Naive ANI -- what you would approximately get without sylph's statistical adjustment (i.e. if you used Mash or Sourmash).</li>
</ol>
<p>Notice the big difference between 1. and 2. This is because the reads are only 1x coverage: <strong>methods like mash screen and sourmash give biased ANI when coverage is low</strong>. </p>
<p>However, the Eff_cov gives smaller than 1x: <strong>this is because Eff_cov takes into account sequencing error</strong>. Sequencing error reduces the k-mer based coverages (sequencing errors invalidate k-mers). You can estimate the true coverage if you provide some more information. See the <a href="../sylph-cookbook/">cookbook</a>.</p>
<p>Here are the ANIs computed by <a href="https://github.com/bluenote-1577/skani">skani</a> between the three genomes:</p>
<div class="highlight"><pre><span></span><code>test_files/e.coli-EC590.fasta<span class="w">   </span><span class="m">100</span>.00<span class="w">  </span><span class="m">99</span>.39<span class="w">   </span><span class="m">98</span>.14
test_files/e.coli-K12.fasta<span class="w"> </span><span class="m">99</span>.39<span class="w">   </span><span class="m">100</span>.00<span class="w">  </span><span class="m">98</span>.09
**test_files/e.coli-o157.fasta<span class="w">  </span><span class="m">98</span>.14<span class="w">   </span><span class="m">98</span>.09<span class="w">   </span><span class="m">100</span>.00**
</code></pre></div>
<p>So the ANIs should be 98.14, 98.09, and 100.0 for EC590, K12, and O157 respectively against the sample. As you can see, Sylph's estimates are quite good and much more reasonable than the Naive ANI.  </p>
<h3 id="profiling-vs-ani-querying-whats-the-difference">Profiling vs ANI querying. What's the difference?</h3>
<p>In the above example, notice that querying each E. coli genome gave a high ANI value. However, only one E. coli genome is present in the sample, not all three.</p>
<p>Thus <code>sylph query</code> is not a <strong>profiler</strong>. It does not tell you <strong>the abundance</strong> of the genomes in your sample, just <strong>how similar</strong> your reference genome is to your metagenome. </p>
<p>To remove this redundancy, we can use the <code>sylph profile</code> instead. </p>
<div class="highlight"><pre><span></span><code>&gt;<span class="w"> </span>sylph<span class="w"> </span>profile<span class="w"> </span>test_files/*<span class="w">  </span><span class="c1"># or use &#39;profile&#39; on the syldb and sylsp.</span>
...
...
Sample_file<span class="w"> </span>Genome_file<span class="w"> </span>Taxonomic_abundance<span class="w"> </span>Sequence_abundance<span class="w">  </span>Adjusted_ANI<span class="w">    </span>Eff_cov<span class="w"> </span>ANI_5-95_percentile<span class="w"> </span>Eff_lambda<span class="w">  </span>Lambda_5-95_percentile<span class="w">  </span>Median_cov<span class="w">  </span>Mean_cov_geq1<span class="w">   </span>Containment_ind<span class="w"> </span>Naive_ANI<span class="w">   </span>Contig_name
../test_files/o157_reads.fastq<span class="w">  </span>../test_files/e.coli-o157.fasta<span class="w"> </span><span class="m">100</span>.0000<span class="w">    </span><span class="m">100</span>.0000<span class="w">    </span><span class="m">99</span>.73<span class="w">   </span><span class="m">0</span>.360<span class="w">   </span><span class="m">99</span>.61-99.91<span class="w"> </span><span class="m">0</span>.360<span class="w">   </span><span class="m">0</span>.33-0.38<span class="w">   </span><span class="m">1</span><span class="w">   </span><span class="m">1</span>.187<span class="w">   </span><span class="m">6208</span>/21899<span class="w">  </span><span class="m">96</span>.02<span class="w">   </span>NZ_CP017438.1<span class="w"> </span>Escherichia<span class="w"> </span>coli<span class="w"> </span>O157:H7<span class="w"> </span>strain<span class="w"> </span><span class="m">2159</span><span class="w"> </span>chromosome,<span class="w"> </span><span class="nb">complete</span><span class="w"> </span>genome
</code></pre></div>
<p>Notice the new <code>Sequence_abundance</code> and <code>Taxonomic_abundance</code> columns, which give the relative abundances as a percentage. See <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8184642/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8184642/</a> for how taxonomic and sequence abundance differ.</p>
<p>There is only 1 genome in the sample, so it has 100% abundance. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This is a simple example with multiple <em>strains</em> of a single <em>species</em>. For real metagenomes, we advise <strong>using databases that are dereplicated at the species level</strong>. </p>
<p>Our <a href="../pre%E2%80%90built-databases/">pre-built databases</a> are all dereplicated at the species level. </p>
</div>
<h3 id="real-profiling-against-a-database">Real profiling against a database</h3>
<p>See the <a href="../taxonomic-profiling-tutorial/">tutorial here</a> to learn how to profile against the GTDB database, an actual database of &gt; 50,000 genomes prokaryotic genomes. </p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.56ea9cef.min.js"></script>
      
    
  </body>
</html>