
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://sylph-docs.github.io/pre%E2%80%90built-databases/">
      
      
        <link rel="prev" href="../install%2Bquickstart/">
      
      
        <link rel="next" href="../sylph-cookbook/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>Pre-built databases - Documentation for sylph - ultrafast, precise metagenomic profiling</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#usage-notes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-header__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Documentation for sylph - ultrafast, precise metagenomic profiling
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Pre-built databases
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Documentation for sylph - ultrafast, precise metagenomic profiling" class="md-nav__button md-logo" aria-label="Documentation for sylph - ultrafast, precise metagenomic profiling" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Documentation for sylph - ultrafast, precise metagenomic profiling
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-can-cannot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    What sylph can and cannot do
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../install%2Bquickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Pre-built databases
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Pre-built databases
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#usage-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Usage notes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage notes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Example usage:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#note-on-taxonomy-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Note on taxonomy usage:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#databases" class="md-nav__link">
    <span class="md-ellipsis">
      Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#parameter-guide" class="md-nav__link">
    <span class="md-ellipsis">
      Parameter Guide
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#database-descriptions" class="md-nav__link">
    <span class="md-ellipsis">
      Database descriptions
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Database descriptions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gtdb-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GTDB Databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#globdb-massive-prokaryotic-catalogue-encompassing-many-other-genome-sets" class="md-nav__link">
    <span class="md-ellipsis">
      GlobDB - massive prokaryotic catalogue encompassing many other genome sets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#other-prokaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Other prokaryotic databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#viral-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Viral databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#eukaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Eukaryotic databases.
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-cookbook/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cookbook
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Output format
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../5%E2%80%90minute-sylph-tutorial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 1 - 5 minute introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../taxonomic-profiling-tutorial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 2 - taxonomic profiling with GTDB
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Taxonomy integration (sylph-tax)
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            Taxonomy integration (sylph-tax)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sylph-tax
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install + Quick start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-commands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Subcommand information
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-output-format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    .sylphmpa output format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sylph-tax-custom-taxonomies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Custom taxonomies
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#usage-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Usage notes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage notes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Example usage:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#note-on-taxonomy-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Note on taxonomy usage:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#databases" class="md-nav__link">
    <span class="md-ellipsis">
      Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#parameter-guide" class="md-nav__link">
    <span class="md-ellipsis">
      Parameter Guide
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#database-descriptions" class="md-nav__link">
    <span class="md-ellipsis">
      Database descriptions
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Database descriptions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gtdb-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GTDB Databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#globdb-massive-prokaryotic-catalogue-encompassing-many-other-genome-sets" class="md-nav__link">
    <span class="md-ellipsis">
      GlobDB - massive prokaryotic catalogue encompassing many other genome sets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#other-prokaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Other prokaryotic databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#viral-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Viral databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#eukaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Eukaryotic databases.
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h2 id="usage-notes">Usage notes</h2>
<p>Pre-sketched databases available for download below. All databases work from sylph version 0.3.x onwards. </p>
<ul>
<li>Use the <strong>Primary</strong> links hosted at <code>http://faust.compbio.cs.cmu.edu</code> if possible. We provide mirrors on google cloud, but this costs us more money.</li>
</ul>
<h4 id="example-usage">Example usage:</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># download database</span>
wget<span class="w"> </span>http://faust.compbio.cs.cmu.edu/sylph-stuff/gtdb-r226-c200-dbv1.syldb

<span class="c1"># profile against database</span>
sylph<span class="w"> </span>profile<span class="w"> </span>gtdb-r226-c200-dbv1.syldb<span class="w"> </span>-1<span class="w"> </span>sample_R1.fq<span class="w"> </span>-2<span class="w"> </span>sample_R2.fq<span class="w">  </span>-t<span class="w"> </span><span class="m">30</span><span class="w"> </span>&gt;<span class="w"> </span>results.tsv
</code></pre></div>
<h3 id="note-on-taxonomy-usage">Note on taxonomy usage:</h3>
<p>Most the databases have associated taxonomies that sylph can utilize. See <a href="../sylph-tax/">here for more information on taxonomy integration</a>.</p>
<h1 id="databases">Databases</h1>
<table>
<thead>
<tr>
<th>Type</th>
<th>Name</th>
<th>Genomes</th>
<th>c-parameter</th>
<th>Size</th>
<th>Primary Download Link</th>
<th>Mirror</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Prokaryotic (GTDB)</strong></td>
<td>GTDB r226</td>
<td>143,614 species</td>
<td>-c 200</td>
<td>18.4 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/gtdb-r226-c200-dbv1.syldb">gtdb-r226-c200-dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/gtdb-r226-c200-dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>GTDB r226</td>
<td>143,614 species</td>
<td>-c 1000</td>
<td>3.7 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/gtdb-r226-c1000-dbv1.syldb">gtdb-r226-c1000-dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/gtdb-r226-c1000-dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>GTDB r220</td>
<td>113,104 species</td>
<td>-c 200</td>
<td>13.1 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/gtdb-r220-c200-dbv1.syldb">gtdb-r220-c200-dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/gtdb-r220-c200-dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>GTDB r220</td>
<td>113,104 species</td>
<td>-c 1000</td>
<td>2.6 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/gtdb-r220-c1000-dbv1.syldb">gtdb-r220-c1000-dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/gtdb-r220-c1000-dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>GTDB r214</td>
<td>85,202 species</td>
<td>-c 200</td>
<td>10 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/v0.3-c200-gtdb-r214.syldb">v0.3-c200-gtdb-r214.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/v0.3-c200-gtdb-r214.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>GTDB r214</td>
<td>85,202 species</td>
<td>-c 1000</td>
<td>2 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/v0.3-c1000-gtdb-r214.syldb">v0.3-c1000-gtdb-r214.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/v0.3-c1000-gtdb-r214.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td><strong>Prokaryotic (GlobDB)</strong></td>
<td>GlobDB r226</td>
<td>306,260 species</td>
<td>-c 200</td>
<td>32 GB</td>
<td>See <a href="https://globdb.org/index.php/home">the GlobDB website</a></td>
<td></td>
<td>Third-party database</td>
</tr>
<tr>
<td></td>
<td>GlobDB r226</td>
<td>306,260 species</td>
<td>-c 1000</td>
<td>6.5 GB</td>
<td>See <a href="https://globdb.org/index.php/home">the GlobDB website</a></td>
<td></td>
<td>Third-party database</td>
</tr>
<tr>
<td><strong>Prokaryotic (Other)</strong></td>
<td>OceanDNA</td>
<td>8,466 ocean MAGs</td>
<td>-c 200</td>
<td>800 MB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/OceanDNA-c200-v0.3.syldb">OceanDNA-c200-v0.3.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/OceanDNA-c200-v0.3.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>SMAG</td>
<td>21,077 soil MAGs</td>
<td>200</td>
<td>2.5 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/SMAG-c200-v0.3.syldb">SMAG-c200-v0.3.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/SMAG-c200-v0.3.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>UHGG v2.0.1 <strong>(not dereplicated)</strong></td>
<td>289,232 gut genomes</td>
<td>-c 200</td>
<td>26 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/uhgg_all_c200_v0.3.0.syldb">uhgg_all_c200_v0.3.0.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/uhgg_all_c200_v0.3.0.syldb">mirror</a></td>
<td>Not dereplicated - do not use for profiling</td>
</tr>
<tr>
<td><strong>Viral</strong></td>
<td>UHGV</td>
<td>171,338 gut vOTUs</td>
<td>-c 100</td>
<td>0.4 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/uhgv_c100_dbv1.syldb">uhgv_c100_dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/uhgv_c100_dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>UHGV</td>
<td>171,338 gut vOTUs</td>
<td>-c 200</td>
<td>0.2 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/uhgv_c200_dbv1.syldb">uhgv_c200_dbv1.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/uhgv_c200_dbv1.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>IMG/VR4.1</td>
<td>2,917,516 viral genomes</td>
<td>-c 200</td>
<td>2 GB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/imgvr_c200_v0.3.0.syldb">imgvr_c200_v0.3.0.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/imgvr_c200_v0.3.0.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td><strong>Eukaryotic</strong></td>
<td>RefSeq Fungi</td>
<td>595 genomes</td>
<td>-c 200</td>
<td>700 MB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/fungi-refseq-2024-07-25-c200-v0.3.syldb">fungi-refseq-2024-07-25-c200-v0.3.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/fungi-refseq-2024-07-25-c200-v0.3.syldb">mirror</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td>TARA Oceans</td>
<td>713 eukaryotic MAGs/SAGs</td>
<td>-c 200</td>
<td>900 MB</td>
<td><a href="http://faust.compbio.cs.cmu.edu/sylph-stuff/tara-eukmags-c200-v0.3.syldb">tara-eukmags-c200-v0.3.syldb</a></td>
<td><a href="https://storage.googleapis.com/sylph-stuff/tara-eukmags-c200-v0.3.syldb">mirror</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="parameter-guide">Parameter Guide</h2>
<ul>
<li><strong>-c 200</strong>: More sensitive, larger file size</li>
<li><strong>-c 1000</strong>: More efficient, smaller file size, less sensitive</li>
<li><strong>-c 100</strong>: More sensitive but primarily for smaller genomes. </li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code>-c 200</code> is used by default, so <code>-c 100</code> must be specified if using a database with -c 100. For example:</p>
<p><code>sylph profile c100_database c1000_database -c100 -1 read1.fq -2 read2.fq</code>. </p>
</div>
<h1 id="database-descriptions">Database descriptions</h1>
<h2 id="gtdb-databases">GTDB Databases</h2>
<p>The <a href="https://gtdb.ecogenomic.org/">GTDB</a> database is a high-quality, curated taxonomy and database for prokaryotes (archaea and bacteria). We take the dereplicated, species-representative genomes (one genome per species). </p>
<p><strong>Available databases</strong>:</p>
<ul>
<li>
<p>GTDB r226 database (143,614 species representative genomes) - April 16, 2025</p>
</li>
<li>
<p>GTDB r220 database (113,104 species representative genomes) - April 24, 2024</p>
</li>
<li>
<p>GTDB r214 database (85,202 species representative genomes) - April 28, 2023</p>
</li>
</ul>
<h2 id="globdb-massive-prokaryotic-catalogue-encompassing-many-other-genome-sets">GlobDB - massive prokaryotic catalogue encompassing many other genome sets</h2>
<p><a href="https://globdb.org/">GlobDB</a> is a catalogue of &gt; 300,000 prokaryotic genomes/MAGs. Their database is dereplicated at ~96% ANI. GlobDB <strong>encompasses 14 other large databases (including GTDB)</strong>. The sylph database is hosted on their website. </p>
<h2 id="other-prokaryotic-databases">Other prokaryotic databases</h2>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>GlobDB encompasses SMAG and a OceanDNA. We <strong>highly recommend using GlobDB if possible</strong> over these databases.  </p>
</div>
<ol>
<li><a href="https://doi.org/10.1038/s41597-022-01392-5">OceanDNA</a> catalogue of 8,466 ocean prokaryotic MAGs, <code>-c 200</code> (800 MB) </li>
<li><a href="https://www.nature.com/articles/s41467-023-43000-z">SMAG</a> catalogue of soil 21,077 soil MAGs, <code>-c 200</code> (2.5 GB):</li>
<li><a href="https://www.ebi.ac.uk/metagenomics/genome-catalogues/human-gut-v2-0-1">UHGG v2.0.1</a> catalogue of 289,232 gut genomes. <strong>Not dereplicated. Do not use for profiling.</strong> <code>-c 200</code> (26 GB): </li>
</ol>
<h2 id="viral-databases">Viral databases</h2>
<ol>
<li><a href="https://github.com/snayfach/UHGV">Unified Human Gut Virome (UHGV) catalog</a> - (171,338 vOTUs from human gut)<ul>
<li>Note: is more refined and has better taxonomic annotations than IMG/VR for human gut</li>
</ul>
</li>
<li>Pre-sketched IMG/VR4.1 database for high-confidence vOTU representatives (2,917,516 viral genomes). </li>
</ol>
<h2 id="eukaryotic-databases">Eukaryotic databases.</h2>
<ol>
<li>595 representative RefSeq fungi genomes (downloaded 2024-07-25), <code>-c 200</code> (700 MB) </li>
<li>713 TARA Oceans eukaryotic MAGs/SAGs from <a href="https://doi.org/10.1016/j.xgen.2022.100123">Delmont et al.</a>, <code>-c 200</code> (900 MB)</li>
</ol>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.56ea9cef.min.js"></script>
      
    
  </body>
</html>